DuckDuckGo/SettingsSubscriptionView.swift

```diff
@@ -244,23 +245,27 @@ struct SettingsSubscriptionView: View {
                         footer: !isSignedIn ? footerLink : nil
                 ) {
 
-                    switch (isSignedIn, hasActiveSubscription, hasNoEntitlements) {
+                    switch (isSignedIn, hasSubscription, hasActiveSubscription, hasAnyEntitlements) {
```

Comment:
To calculate the state the 4th bool is added to tell if there is subscription present at all.

Reply:
The order of cases was done in reverse as it is more logical to read them incrementally and check if all cases are covered.